tests/python_tests/test_sampling.py

```diff
@@ -317,6 +317,64 @@ def test_individual_generation_configs_random(tmp_path, test_struct: RandomSampl
     # Discrete_distribution impl depends on platform, model inference results may depend on CPU.
 
 
+@pytest.mark.precommit
+@pytest.mark.parametrize("sampling_config", [get_greedy(), get_beam_search(), get_multinomial_all_parameters()])
+@pytest.mark.parametrize("max_num_batched_tokens", [2, 4, 256])
+def test_echo_without_completion(tmp_path, sampling_config, max_num_batched_tokens):
+    generation_config = sampling_config
```

Comment:
It python it's just reference assignment. You probably need `copy.deepcopy()`.

Reply:
Even if the test is parametrized? It's called 9 times and I suppose every time we get new sampling_config so even if it's just a reference isn't it okay?

Reply:
You shouldn't introduce one extra reference at all in that case.